src/tpm2_pytss/TCTISPIHelper.py

```diff
@@ -0,0 +1,296 @@
+            missing_implementation.append("on_spi_transfer")
+
+        if "TCTISPIHelper.on_timeout_expired" in str(self.on_timeout_expired):
+            missing_implementation.append("on_timeout_expired")
+
+        if "TCTISPIHelper.on_start_timeout" in str(self.on_start_timeout):
+            missing_implementation.append("on_start_timeout")
+
+        if "TCTISPIHelper.on_sleep_ms" in str(self.on_sleep_ms):
+            missing_implementation.append("on_sleep_ms")
+
+        if len(missing_implementation) > 0:
+            raise NotImplementedError(
+                f"Subclasses must implement {','.join(missing_implementation)}"
+            )
+
+        _chkrc(lib.Tss2_Tcti_Spi_Helper_Init(ffi.NULL, size, self._callbacks))
+
+        self._tcti_mem = ffi.new(f"uint8_t [{size[0]}]")
+        self._opaque_tcti_ctx = ffi.cast("TSS2_TCTI_CONTEXT *", self._tcti_mem)
+
+        try:
+            self._clear_exceptions()
+            _chkrc(
+                lib.Tss2_Tcti_Spi_Helper_Init(
+                    self._opaque_tcti_ctx, size, self._callbacks
+                )
+            )
+        except Exception as e:
+            e = self._get_current_exception(e)
+            self._clear_exceptions()
+            raise e
+
+        super().__init__(self._opaque_tcti_ctx)
+
+    @property
+    def waitstate(self):
+        """Gets the wait state property.
+
+        Returns(bool):
+            True if this TCTI implements wait states, false otherwise.
+        """
+        return self._with_wait_state
+
+    @staticmethod
+    def _cffi_cast(userdata):
+        return ffi.from_handle(userdata)
+
+    def on_sleep_ms(self, milliseconds: int) -> None:
+        """Sleeps for a specified amount of time in millisecons.
+
+        This callback is REQUIRED.
+        No errors may occur across this boundary.
+
+        Args:
+            milliseconds(int): The time to sleep.
+        """
+        pass
+
+    def on_start_timeout(self, milliseconds: int) -> None:
+        """Called when a timeout is occurring with the sleep duration in millisecons.
+
+        This callback is REQUIRED.
+        No errors may occur across this boundary.
+
+        Args:
+            milliseconds(int): The time to sleep.
+        """
+        pass
+
+    def on_timeout_expired(self) -> bool:
+        """Called to determine if a timeout is expired.
+
+        This callback is REQUIRED.
+        No errors may occur across this boundary.
+        """
+        pass
+
+    def on_spi_transfer(self, data_in: bytes) -> bytes:
+        """Called to transfer data across the SPI bus.
+
+        This callback is REQUIRED.
+        No errors may occur across this boundary.
+
+        Args:
+            data_in(bytes): The data to send.
+
+        Returns(bytes):
+            The bytes to send.
+
+        Raises:
+            TSS2_Exception: if an error occurs and the RC is returned to the
+                native TCTI.
+        """
+
```

Comment:
Missing pass

Reply:
I guess pass isn't needed since they have a docstring, but I guess Ill add it to keep it consistent or should I just delete them?